### PR TITLE
Feat/icp as standard token

### PR DESCRIPTION
**Pull request by angarita-dev**
# ⚠️ DO NOT MERGE, UPDATE & TEST WITH PROVIDER
## Changelog

- Treats `ICP` as a standard token
- Adds `amount` parsing from `plug-controller`
- Fixes bug on activity

### Type of changes included:

- [x] Components
- [x] Business Logic
- [x] Bug fixes
- [ ] Styling
- [ ] Code Refactor

### Clubhouse Tickets Related:

- Ticket 1
- Ticket 2
- Ticket 3

### Screenshots/Gifs:

*Place you changes screenshots here*

### Notes:

*Place special notes here*

### Tested on:

- [x] Chrome
- [ ] Firefox
- [ ] Safari

### CODE REVIEW

**Review by aleacevedo — APPROVED** [review]

File: source/Background/Keyring.js [hunk 1]
```diff
@@ -88,12 +134,14 @@ export const getKeyringHandler = (type, keyring) => ({
   [HANDLER_TYPES.UNLOCK]: async (params) => {
     let unlocked = false;
     try {
+      console.log('trying to unlock');
```
Comment (rocky-fleek) [inline source/Background/Keyring.js:137]: remove logs

File: source/Background/Keyring.js [hunk 2]
```diff
@@ -117,33 +165,40 @@ export const getKeyringHandler = (type, keyring) => ({
...
     try {
       if (!keyring?.isUnlocked) return {};
 
       const { wallets, currentWalletId } = await keyring.getState();
-      let assets = wallets?.[currentWalletId]?.assets;
-      const shouldUpdate = assets?.every((asset) => !Number(asset.amount))
-        || assets?.some((asset) => asset.amount === 'Error')
+      let assets = Object.values(wallets?.[currentWalletId]?.assets);
+      const shouldUpdate = Object.values(assets)?.every((asset) => !Number(asset.amount))
+        || Object.values(assets)?.some((asset) => asset.amount === 'Error')
```
Comment (rocky-fleek) [inline source/Background/Keyring.js:177]: I'm afraid these kinda changes might break the extension on the first render. But we should do upgrade tests for this

Comment (angarita-dev) [reply source/Background/Keyring.js:177]: I think on the first render we might have the balances broken with the manual refresh button, but after the first update we shouldn't have any issues

File: source/shared/constants/currencies.js [hunk 3]
```diff
@@ -100,9 +100,18 @@ export const parseToAmount = (amount, decimals) => {
...
+
+  if (shouldParseDecimalExponential > -1) {
+    const decimalsToAdd = stringifiedAmount.slice(shouldParseDecimalExponential + 2) - 1;
+    const prefix = stringifiedAmount.slice(0, shouldParseDecimalExponential);
+    stringifiedAmount = `0.${'0'.repeat(decimalsToAdd)}${prefix}`;
+  }
+
+  const commaIndex = stringifiedAmount.search(/[.]/);
+  const decimalsToFill = decimals - (stringifiedAmount.length - commaIndex - 1);
+  const strippedText = stringifiedAmount.slice(0, commaIndex) + stringifiedAmount.slice(commaIndex + 1) + '0'.repeat(decimalsToFill);
```
Comment (tomiir) [inline source/shared/constants/currencies.js:114]: :monkapray:

**Review by tomiir — APPROVED** [review]
Update controller ref and it lgtm

**Pull request merged**